sdk/src/trace/tracer.h

```diff
@@ -14,10 +14,12 @@ namespace trace
 class Tracer final : public trace_api::Tracer, public std::enable_shared_from_this<Tracer>
 {
 public:
-  // Note: recorder must be non-null
-  explicit Tracer(std::unique_ptr<Recorder> &&recorder) noexcept : recorder_{std::move(recorder)} {}
+  // Note: processor must be non-null
+  explicit Tracer(std::unique_ptr<SpanProcessor> &&processor) noexcept
+      : processor_{std::move(processor)}
+  {}
 
-  Recorder &recorder() const noexcept { return *recorder_; }
+  SpanProcessor &processor() const noexcept { return *processor_; }
```

Comment:
My understanding is that `Tracer` doesn't have `SpanProcessor`?

Reply:
I now went with the approach we discussed today in the SIG: the tracer holds a span processor that is updated when calling `SetProcessor` on the tracer provider.